04.ls/ls.rb

```diff
@@ -47,32 +53,27 @@ def output(file_names, directory_path, options)
 end
 
 def output_long_listing_format(file_names, directory_path)
-  puts "total #{calc_block_count_total(file_names, directory_path)}"
-
   details_by_file_name = build_details_by_file_name(file_names, directory_path)
 
-  max_width_by_detail = calc_max_width_by_detail(details_by_file_name)
+  puts "total #{calc_block_count_total(details_by_file_name)}"
 
-  file_names.each do |file_name|
-    details = details_by_file_name[file_name]
+  max_width_by_detail = calc_max_width_by_detail(details_by_file_name)
 
-    DETAILS_OUTPUT_ORDER.each do |key|
-      if RJUST_LIST.include?(key)
-        print details[key].rjust(max_width_by_detail[key])
-      else
-        print details[key].ljust(max_width_by_detail[key])
-      end
+  details_by_file_name.each do |file_name, details|
+    DETAILS_KEYS.each do |key, align|
+      value = details[key]
+      width = max_width_by_detail[key]
+      print(align == :right ? value.rjust(width) : value.ljust(width))
       print ' '
     end
     puts file_name
   end
 end
 
-def calc_block_count_total(file_names, directory_path)
-  file_names.map do |file_name|
-    file_path = "#{directory_path}/#{file_name}"
-    File.stat(file_path).blocks
-  end.sum
+def calc_block_count_total(details_by_file_name)
+  details_by_file_name.sum do |_file_name, details|
```

Comment:
思ったんですが、`details_by_file_name`ってハッシュである必然性はなくて、`details`の中にnameも含めてしまって、「ハッシュが各要素になっている配列」にしたらいいのでは？と思いました。

```suggestion
  details_by_file_name.sum do |details|
```

Reply:
気付いていませんでしたが、その通りですね。
 `file_names`と`details`を別のものとして扱う、という考えで固まってしまっていました。
 
 おかげさまで入れ子構造がなくなりわかりやすくなりました。